src/hooks/useFetch.ts

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useReducer, useRef } from 'preact/hooks';
+
+// Adapted after: https://usehooks-ts.com/react-hook/use-fetch
+
+type Options = {
+    enableCache?: boolean;
+};
+
+interface State<T> {
+    data?: T;
+    error?: Error;
+}
+type Cache<T> = { [url: string]: T };
+type Action<T> = { type: 'loading' } | { type: 'fetched'; payload: T } | { type: 'error'; payload: Error };
+
+export const useFetch = <T = unknown>(url?: string, options?: Options, fetch_options?: RequestInit): State<T> => {
```

Comment:
I think the order of parameters should be the other way around. It is more likely to leave the `useFetch` specific options empty and that would be annoying.

```suggestion
export const useFetch = <T = unknown>(url?: string, fetch_options?: RequestInit, options?: Options): State<T> => {
```

Reply:
I feel like your new implementation is a little clumsy… Don't we use the `fetchOptions` on nearly every request?

Reply:
Do we? I just looked through all our uses of `fetch()`. Of the ones where it would make sense to replace them with this, none set any options.

There are only three instances where we set options but those are all `POST`/`PUT` and I definitely wouldn't use `useFetch` for that.